gmapping/src/tftest.cpp

```diff
@@ -1,16 +1,16 @@
 #include <cstdio>
 #include "sensor_msgs/LaserScan.h"
 #include "ros/node.h"
-#include "tf/transform_listener.h"
+#include <tf2_ros/transform_listener.h>
 
 class Test
 {
   public:
     Test()
     {
       node_ = new ros::Node("test");
-      tf_ = new tf::TransformListener(*node_, true, 
-                                      ros::Duration(10));
+      tf_.reset(new tf2::Buffer(ros::Duration(10)));
+      tfl_ = new tf2_ros::TransformListener(*tf_, *node_, true);
       tf_->setExtrapolationLimit( ros::Duration().fromSec(0.2));
```

Comment:
setExtrapolationLimit was chosen not to be implemented in tf2 as there wasn't yet found a good use case for it in tf. I'm not sure that this unit will compile.